### DISCUSSION TITLE
Jekyll Image slider in al-folio webpage!

### DISCUSSION THREAD

**Discussion by mihir022** [alshedivat/al-folio#1815] [Q&A] [answered]
I am using this template to make a website for our lab. I want to add an **Image Slider** in the About page just below the description. What process or required steps should I follow to add js enabled image slider? Can you tell me what changes should I do in which particular files to reflect the slider on the About page? 
Here is demo of slider [Image Slider](https://jekylltools.github.io/jekyll-ideal-image-slider-include/examples/).

**Comment by george-gca** [comment]
Have you followed the steps in their repo [README](https://github.com/jekylltools/jekyll-ideal-image-slider-include)?

PS: it might simply not work, since the repo hasn't been updated in years. Also it has this information in the repo:

> This repository is archived. Issues are disabled. Pull requests will be ignored.

If you want to try it, probably the files that you'll need to modify in our code are `_pages/about.md` and `_layouts/default.html`.

**Reply by mihir022** [reply]
I have read and tried all the steps from repository but still I got no output. Are there any other way to work around this archived repository?

**Comment by pskadasi** [comment]
I am also trying to do the same, but I am not able to arrive at the solution. can anyone kindly help me on this thing?

**Reply by pskadasi** [reply]
this is resolved, thanks all.

**Reply by george-gca** [reply]
Can you share your solution, so others may benefit from it?

**Reply by pskadasi** [reply]
sure, i'll get back in 3-4 days and post the workaround.

**Comment by pskadasi** [comment, marked as answer]
as @george-gca said, i tried to follow Readme: https://github.com/jekylltools/jekyll-ideal-image-slider-include

but it would be better if we follow gh_pages branch and try to replicate what they have done for slider: https://github.com/jekylltools/jekyll-ideal-image-slider-include/tree/gh-pages

even after you follow all the steps, you might get a javascript error, that is due to slider_scripts.html file here _includes/slider_scripts.html

line 12-14:

```
<script src="{{ "/assets/js/slider/ideal-image-slider.min.js" | relative_url }}"></script>
<script src="{{ "/assets/js/slider/iis-bullet-nav.js" | relative_url }}"></script>
<script src="{{ "/assets/js/slider/iis-captions.js" | relative_url }}"></script>
```

replace the above lines with the below:


```
<script src="{{ '/assets/js/slider/ideal-image-slider.min.js' | relative_url | bust_file_cache }}"></script>
<script src="{{ '/assets/js/slider/iis-bullet-nav.js' | relative_url | bust_file_cache }}"></script>
<script src="{{ '/assets/js/slider/iis-captions.js' | relative_url | bust_file_cache }}"></script>
```

basically replacing " with ' solved my error, and slider worked fine. and replace all other above script lines with new script lines in other files as well.

let me know if anyone is facing the problem even if this doesn't solve for you.

**Reply by mpil-gist** [reply]
It works well. Thanks!

**Comment by scaliaven** [comment]
Is there any way to add an image slider to the profile image in the About page?

**Reply by george-gca** [reply]
Yes, we have an example in [this blog post](https://alshedivat.github.io/al-folio/blog/2024/advanced-images/).

**Reply by scaliaven** [reply]
I think this is mainly for pictures in blog posts, not sure whether it works for profile images in the About page in a similar way.

**Reply by george-gca** [reply]
It should work. The `about` page uses the `about` layout

https://github.com/alshedivat/al-folio/blob/ac99061bceba298794cacb994dbc31f1b5474559/_pages/about.md?plain=1#L2

which uses the `default` layout

https://github.com/alshedivat/al-folio/blob/ac99061bceba298794cacb994dbc31f1b5474559/_layouts/about.liquid#L2

which includes all javascript code for the libraries

https://github.com/alshedivat/al-folio/blob/ac99061bceba298794cacb994dbc31f1b5474559/_layouts/default.liquid#L54

**Reply by scaliaven** [reply]
Thanks, will try that. However, one concern I have is that the about page's profile image is defined inside the frontmatter, whereas in the blog post, advanced image features work after the front matter is properly set.

**Reply by george-gca** [reply]
For your case you'll have to edit your `about` layout, changing the code that handles the profile by one with the slider. I meant that the needed libraries are already imported there, as long as you set it on the front matter, so it won't be hard to add it:

```yml
images:
  slider: true
```

**Reply by scaliaven** [reply]
That makes sense, thanks.
